src/Compiler/Utilities/range.fsi

```diff
@@ -102,6 +102,9 @@ type Range =
 
     /// The file name for the file of the range
     member FileName: string
+    
+    /// Apply the line directives to the range.
+    member ApplyLineDirectives: unit -> range
```

Comment:
The signature of this is what I had in mind.
Maybe the input (`this`) range should be a different type compared to the output ?